include/atomic.cuh

```diff
@@ -48,8 +50,10 @@ static inline __device__ double atomicAdd(double* address, double val)
 */
 static inline __device__ double2 atomicAdd(double2 *addr, double2 val){
   double2 old = *addr;
+#if defined(__CUDA_ARCH__) || CUDA_VERSION >= 8000
```

Comment:
Why is that function just doing nothing on CUDA 7.5? Is that safe?

Reply:
It doesn't do nothing on CUDA 7.5, since it is also activated if `__CUDA_ARCH__` is defined.  The issue is here that clang-cuda will not allow us to provide our own definition of `atomicAdd` since it sees the definition in the header which was provided for Pascal and above on CUDA 8.0 and above.  This is a restriction of how clang and nvcc resolve functions marked as `__device__` / `__host__`.

The end result here is that on clang-cuda we cannot provide support on Kepler architectures (which require a homegrown version of `atomicAdd` using `atomicCAS`) without using a different name for `atomicAdd` or applying some hacks to the CUDA headers.  Since Kepler support is on the way out for perhaps QUDA 2.0, I don't see this as a major issue.

Reply:
Makes sense now. Thanks. 
Maybe worth adding a comment so this doesn’t confuse anyone in the future.